### PR TITLE
Issue 457 add explanations

### PR DESCRIPTION
**Pull request by haim-kermany**
three major things:
1. add tooltip for each element
2. add explanation for each subgraph
3. give meaningful names to the connectivity shorts

### CODE REVIEW

File: nca/FWRules/DotGraph.py [hunk 1]
```diff
@@ -159,9 +192,11 @@ def _node_to_str(self, node):
                     table += f'<tr><td>{line}</td></tr>'
             table += '</table>>'
             label = f'label={table}'
-            return f'\t\"{node.name}\" [{label} {self.node_styles[node.node_type]}]\n'
+            node_desc = f'{label} {self.node_styles[node.node_type]} tooltip=\"{self.node_tooltip[node.node_type]}\"'
         else:
-            return f'\t\"{node.name}\" [{self.node_styles[node.node_type]}  xlabel=\"{self.labels_dict[node.label[0]]}\"]\n'
+            node_desc = f'{self.node_styles[node.node_type]}  xlabel=\"{self.labels_dict[node.label[0]]}\" ' \
+                   f'tooltip=\"{self.node_tooltip[node.node_type]}{node.label[0]}\"'
```
Comment (zivnevo) [inline nca/FWRules/DotGraph.py:198]: Why do we add now `{node.label[0]}`?

File: nca/FWRules/DotGraph.py [hunk 2]
```diff
@@ -190,26 +226,31 @@ def _set_labels_dict(self):
...
-                # we want sort the labels before giving each label its index:
-                token_labels.sort()
-                for label in token_labels:
-                    self.labels_dict[label] = f'{first_token}_{token_labels.index(label)}'
-                    # todo - maybe put another token instead of the index
-                    # we should revisit this code after reformatting connections labels
+                # we want sort the labels before giving each label its letter:
+                short_labels.sort()
+                for label in short_labels:
+                    self.labels_dict[label] = f'{short}{string.ascii_lowercase[short_labels.index(label)]}'
```
Comment (zivnevo) [inline nca/FWRules/DotGraph.py:255]: I'm not sure I understand what this code does. What happens if `short_labels.index(label) > 26`?

**Review by zivnevo — CHANGES_REQUESTED** [review]
Some comments to start with. I didn't review `InteractiveConnectivityGraph.py` yet.

File: requirements.txt [hunk 3]
```diff
@@ -1,5 +1,5 @@
 ghapi==1.0.3
 PyYAML==6.0
 greenery==4.0.0
-networkx==3.1
+networkx==3.0
```
[the remark this answers is not in the document]
Comment (haim-kermany) [reply requirements.txt:4]: blame it on the git
fixing

File: nca/FWRules/DotGraph.py [hunk 4]
```diff
@@ -97,14 +107,36 @@ def to_str(self):
...
+        output_result += '\tcolor=white\n'
+        output_result += self._explanation_to_str()
+        output_result += '\tlabelloc = "b"\n'
+        output_result += '\tfontsize=15\n'
+        output_result += '\tfontcolor=maroon\n'
+        output_result += '\t}\n'
         output_result += '}\n'
         return output_result
 
+    def _explanation_to_str(self):
```
[the remark this answers is not in the document]
Comment (haim-kermany) [reply nca/FWRules/DotGraph.py:125]: fixing

File: nca/FWRules/DotGraph.py [hunk 5]
```diff
@@ -97,14 +107,36 @@ def to_str(self):
...
+        """
+        creates a string in dot format og the explanation label
```
[the remark this answers is not in the document]
Comment (haim-kermany) [reply nca/FWRules/DotGraph.py:127]: fixing

File: nca/FWRules/DotGraph.py (hunk 1 above)
Comment (haim-kermany) [reply nca/FWRules/DotGraph.py:198]: in bi/clique, the node label is the connectivity, we add it also to the tool tip

File: nca/FWRules/DotGraph.py (hunk 2 above)
Comment (haim-kermany) [reply nca/FWRules/DotGraph.py:255]: its a bug, fixing.

File: nca/FWRules/DotGraph.py (hunk 1 above)
Comment (zivnevo) [reply nca/FWRules/DotGraph.py:200]: got it

**Review by zivnevo — APPROVED** [review]

**Pull request merged**